tokio/src/net/tcp/listener.rs

```diff
@@ -234,12 +281,26 @@ impl TcpListener {
     /// from a future driven by a tokio runtime, otherwise runtime can be set
     /// explicitly with [`Runtime::enter`](crate::runtime::Runtime::enter) function.
     #[track_caller]
-    pub fn from_std(listener: net::TcpListener) -> io::Result<TcpListener> {
+    pub fn from_tcp_unchecked(listener: net::TcpListener) -> io::Result<TcpListener> {
         let io = mio::net::TcpListener::from_std(listener);
         let io = PollEvented::new(io)?;
         Ok(TcpListener { io })
     }
 
+    /// Creates new `TcpListener` from a `std::net::TcpListener`.
+    ///
+    /// Deprecated because easy to misuse and naming doesn't warn enough about it
+    /// (you may want to favor using
+    /// [`from_tcp`](TcpListener::from_tcp) instead of
+    /// [`from_tcp_unchecked`](TcpListener::from_tcp_unchecked))
+    ///
+    /// This has the same behavior as [`TcpListener::from_tcp_unchecked`].
+    #[track_caller]
+    #[deprecated = "Easy to misuse - use from_tcp or from_tcp_unchecked instead"]
+    pub fn from_std(listener: net::TcpListener) -> io::Result<TcpListener> {
```

Comment:
The grammar here has some issues. Please make sure that your sentences have a subject. Furthermore, it's best to avoid parenthetical statements.